server/eventHandlers/notifications.ts

```diff
@@ -56,7 +58,39 @@ export default class extends IEventHandler {
         event.excludeAddresses,
         event.includeAddresses
       );
+
+      const promises = dbNotifications.map(
+        (x) => {
+          // add chainEventType to published message since chainEventType is used to determine room to emit to
+          x = JSON.parse(JSON.stringify(x));
+          x.ChainEvent = event;
+          x.ChainEvent.ChainEventType = dbEventType;
+          return this._rabbitMqController.publish(x, 'ChainEventsNotificationsPublication')
+        }
+      )
+      
+      // polyfill
```

Comment:
Can you explain this block pls?

Reply:
Thanks for asking about this - it works but is somewhat incorrect so I will push a fix tomorrow that will clarify this.

Reply:
This turned into quite a large thing to fix. The problem is all notifications atm revolve around the notification id for marking is_read. The problem with this is if we want to publish the same chain events to multiple users the notification ids can't all be the same (i.e. each user gets their own notification for a chain event with its individual id). I'm working on a separate PR right now that would separate the notification data from the subscription/user. So in other words a chain event will have 1 notification id that can be referenced by all users (rather than a new notification id for every user). This allows us to send the same message to multiple users at the same time much more efficiently. To mark a notification as is_read we just match the subscription id and notification id in the new Notifications_Read table. Might be confusing written here but I will clarify in the PR.

Reply:
Sounds good -- you should consult with Jesse about table design before moving forward. Might be a good opportunity to clean up how notifications work in the DB.